server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -310,12 +310,12 @@ private static void innerParseObject(DocumentParserContext context, ObjectMapper
             switch (token) {
                 case FIELD_NAME:
                     currentFieldName = parser.currentName();
+                    if (currentFieldName.isEmpty()) {
```

Comment:
I honestly don't know, I figured that maybe that's not needed because there are no params, but I am really guessing. @original-brownbear what do you think?

Reply:
I think there's a tiny advantage to having a separate method (or at least was last time I measured this kind of thing) because eventually the JVM figures out that the method call to the throwing method is dead and removes the inlined method => code size wise a call to a single static method is smaller than throw + constructor call.
Probably doesn't matter too much here since the method is quite big already.